proto/substrait/relations.proto

```diff
@@ -127,11 +127,20 @@ message JoinRel {
     JOIN_TYPE_RIGHT = 4;
     JOIN_TYPE_SEMI = 5;
     JOIN_TYPE_ANTI = 6;
+    JOIN_TYPE_SINGLE=7;
   }
 
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+message CrossRel {
```

Comment:
Does this need to be a separate relation from `Join` or can it be another join variant? No strong opinion on that from me, just curious if there's a reason for it to be a distinct relation.

Reply:
If there is no requirement of a join expression, this could be a cross_join type. I'm also fine with that.

Reply:
I see, maybe it makes sense to keep this as you have it then, so that a plan must explicitly be encoded with cross join and we don't have to deal with cross joins with filters.

Reply:
I'm fine either way on this. I think there are some nice properties to having crossrel separate but it will probably look very similar to join ultimately.